### PR TITLE
Create a new permission to limit the scope for RoleUploadOnly

### PR DESCRIPTION
**Pull request by kpom-specter**
…

## Description
Create a new permission that controls the ingestion/file upload routes, and add that permission to the roles that currently have GraphDBWrite (and remove that permission where necessary)

## Motivation and Context

This PR addresses: BED-4903 and BED-4950

The permission intended to control upload rights for clients should only be allowed to perform uploads. The GraphDBWrite permission also includes to modify access groups and trigger manual analysis runs.

## How Has This Been Tested?

*Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc.*

## Types of changes

- Bug fix (non-breaking change which fixes an issue)
- Database Migrations

## Checklist:

- [x] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [ ] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [x] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

**Comment by kpom-specter** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cmd/api/src/auth/role_test.go [hunk 1]
```diff
@@ -185,12 +185,12 @@ func testRoleAccess(t *testing.T, roleName string) {
 			}
 		}),
 
-		lab.TestCase(fmt.Sprintf("%s be able to access GraphDBWrite endpoints", testCondition(role, auth.Permissions().GraphDBWrite)), func(assert *require.Assertions, harness *lab.Harness) {
+		lab.TestCase(fmt.Sprintf("%s be able to access GraphDBIngest endpoints", testCondition(role, auth.Permissions().GraphDBIngest)), func(assert *require.Assertions, harness *lab.Harness) {
```
Comment (mistahj67) [inline cmd/api/src/auth/role_test.go:188]: As we've swapped this to GraphDBIngest, would you mind adding a GraphDBWrite test to this so we continue to have that coverate?

File: cmd/api/src/auth/role.go [hunk 2]
```diff
@@ -56,7 +56,7 @@ func Roles() map[string]RoleTemplate {
 			Description: "Used for data collection clients, can post data but cannot read data",
 			Permissions: model.Permissions{
 				permissions.ClientsTasking,
-				permissions.GraphDBWrite,
+				permissions.GraphDBIngest,
```
[the remark this answers is not in the document]
Comment (mistahj67) [reply cmd/api/src/auth/role.go:59]: Yep, the permission as well as a migration tying the role to the permission needs to be added. There's examples in other migrations for generic sql formatting. Don't forget to include an Administrator role sql statement as well.

[the remark this answers is not in the document]
Comment (kpom-specter) [reply cmd/api/src/auth/role.go:59]: 👍

File: cmd/api/src/auth/role_test.go (hunk 1 above)
Comment (kpom-specter) [reply cmd/api/src/auth/role_test.go:200]: Done. This was the trickiest part 😅

File: cmd/api/src/database/analysisrequest.go [hunk 3]
```diff
@@ -86,7 +86,7 @@ func (s *BloodhoundDB) setAnalysisRequest(ctx context.Context, requestType model
 	} else {
 		// Analysis request existed, we only want to overwrite if request is for a deletion request, otherwise ignore additional requests
 		if analReq.RequestType == model.AnalysisRequestAnalysis && requestType == model.AnalysisRequestDeletion {
-			updateSql := `update analysis_request_switch set requested_by = ?, request_type = ?, requested_at = ? limit 1;`
+			updateSql := `update analysis_request_switch set requested_by = ?, request_type = ?, requested_at = ?;`
```
Comment (mistahj67) [inline cmd/api/src/database/analysisrequest.go:89]: 👀 Did this sneak in from main or is this intentional here?

File: cmd/api/src/database/migration/migrations/v6.2.0.sql [hunk 4]
```diff
@@ -24,3 +24,28 @@ VALUES (current_timestamp,
...
+        (SELECT id FROM permissions WHERE permissions.authority = 'graphdb' and permissions.name = 'Ingest'))
+ON CONFLICT DO NOTHING;
+
+-- Grant the Admininstrator user GraphDBIngest permissions
+INSERT INTO roles_permissions (role_id, permission_id)
+VALUES ((SELECT id FROM roles WHERE roles.name = 'Administrator'),
+        (SELECT id FROM permissions WHERE permissions.authority = 'graphdb' and permissions.name = 'Ingest'))
+ON CONFLICT DO NOTHING;
+
+-- Remove the GraphDBWrite permission from the Upload-Only role for
```
Comment (mistahj67) [inline cmd/api/src/database/migration/migrations/v6.2.0.sql:48]:
```suggestion
-- Remove the GraphDBWrite permission from the Upload-Only role
```

**Review by ddlees — APPROVED** [review]

**Review by mistahj67 — APPROVED** [review]
Looks good to me,  please verify that the analysisrequested db sql method change was intentional before merging though! 🚀

File: cmd/api/src/database/analysisrequest.go (hunk 3 above)
Comment (kpom-specter) [reply cmd/api/src/database/analysisrequest.go:89]: This is also addressing this ticket as I stumbled upon it. https://specterops.atlassian.net/browse/BED-4950

I'll link the ticket in the description (or I can take this out and make a separate MR if that's more appropriate)

**Review by elikmiller — APPROVED** [review]
Validated migration, tests pass, manually tested new permissions. Looks good!

**Pull request merged**